lib/src/Uncertainty/Distribution/Hypergeometric.cxx

```diff
@@ -0,0 +1,443 @@
+//                                               -*- C++ -*-
+/**
+ *  @brief The Hypergeometric distribution
+ *
+ *  Copyright 2005-2019 Airbus-EDF-IMACS-Phimeca
+ *
+ *  This library is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+#include <cmath>
+#include "openturns/Hypergeometric.hxx"
+#include "openturns/SpecFunc.hxx"
+#include "openturns/DistFunc.hxx"
+#include "openturns/RandomGenerator.hxx"
+#include "openturns/PersistentObjectFactory.hxx"
+#include "openturns/Exception.hxx"
+
+BEGIN_NAMESPACE_OPENTURNS
+
+CLASSNAMEINIT(Hypergeometric)
+
+static const Factory<Hypergeometric> Factory_Hypergeometric;
+
+/* Default constructor */
+Hypergeometric::Hypergeometric()
+  : DiscreteDistribution()
+  , n_(0)
+  , k_(0)
+  , m_(0)
+{
+  setName( "Hypergeometric" );
+  // We set the dimension of the Hypergeometric distribution
+  setDimension( 1 );
+  computeProbabilities();
```

Comment:
The default parametrization n=0 k=0 m=0 doesn't yield a valid distribution and crashes in computeProbabilities at line 355 : `probabilities_[kMode] = p;`.